merlin/dag/executors.py

```diff
@@ -48,110 +48,114 @@ def transform(
         Transforms a single dataframe (possibly a partition of a Dask Dataframe)
         by applying the operators from a collection of Nodes
         """
-        output_df = None
+        output_data = None
 
         for node in nodes:
-            input_df = self._build_input_data(node, df, capture_dtypes=capture_dtypes)
+            input_data = self._build_input_data(node, transformable, capture_dtypes=capture_dtypes)
 
             if node.op:
-                transformed_df = self._transform_data(node, input_df, capture_dtypes=capture_dtypes)
+                transformed_data = self._transform_data(
+                    node, input_data, capture_dtypes=capture_dtypes
+                )
             else:
-                transformed_df = input_df
+                transformed_data = input_data
 
-            output_df = self._combine_node_outputs(node, transformed_df, output_df)
+            output_data = self._combine_node_outputs(node, transformed_data, output_data)
 
         if additional_columns:
```

Comment:
this is a more general question that maybe doesn't affect this particular change, however it seems related enough to ask here: What is this `additional_columns` functionality for in the `LocalExecutor`? I can see it being used as part of the transform in the `DaskExecutor`. Would it simplify things if we dropped that here or is there a use-case not covered by the tests?

Reply:
`additional_columns` is a way for nodes in the graph to fetch columns from the input data that are not transformed by the parent or dependency nodes in the graph. It's used in `DaskExecutor.fit()` when forming the transformed version of the data from previous nodes in order to compute statistics over it in order to merge in columns that don't need to be transformed by the upstream nodes. In order to do that, it gets passed into `DaskExecutor.transform()`, which then delegates the transformation of each Dask DataFrame partition to the `LocalExecutor.transform()`.